ffcx/codegeneration/integrals_generator.py

```diff
@@ -149,12 +148,49 @@ def generate(self):
                           "coordinate_dofs = (const double*)__builtin_assume_aligned(coordinate_dofs, {});"
                           .format(alignment))]
 
+        self.contains_reflections = {}
+        c_false = L.LiteralBool(False)
         for element, id in self.ir.element_ids.items():
-            parts += get_vector_reflection_array(L, self.ir.element_dof_reflection_entities[element],
-                                                 "ref_dof" + str(id))
-
+            vname = "ref_dof" + str(id)
+            self.contains_reflections[id] = False
+            reflect_dofs = []
+            for dre in self.ir.element_dof_reflection_entities[element]:
+                if dre is None:
+                    # Dof does not need reflecting, so put false in array
+                    reflect_dofs.append(c_false)
+                else:
+                    # Loop through entities that the direction of the dof depends on to
+                    # make a conditional
+                    ref = c_false
+                    for j in dre:
+                        if ref == c_false:
+                            # No condition has been added yet, so overwrite false
+                            ref = self.backend.symbols.get_reflection(L, j)
+                        else:
+                            # This is not the first condition, so XOR
+                            ref = L.Conditional(self.backend.symbols.get_reflection(L, j), L.Not(ref), ref)
+                    reflect_dofs.append(ref)
+                    if ref != c_false:
+                        # Mark this space as needing reflections
+                        self.contains_reflections[id] = True
+
+            # If no dofs need reflecting, don't write any array
+            if self.contains_reflections[id]:
+                parts.append(L.ArrayDecl(
+                    "const bool", L.Symbol(vname), (len(reflect_dofs), ), values=reflect_dofs))
+
+        self.table_dofmaps = {}
         for tname, dofmap in self.ir.table_dofmaps.items():
```

Comment:
Similar here, make `IntegralGenerator.generate_table_dofmaps` method to keep the same code granularity.